bugsnag_test.go

```diff
@@ -1,19 +1,65 @@
 package bugsnag
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"net"
 	"net/http"
+	"net/http/httptest"
 	"strings"
-	"sync"
 	"testing"
 	"time"
 
 	"github.com/bitly/go-simplejson"
+	"github.com/bugsnag/bugsnag-go/sessions"
 )
 
+// The line numbers of this method are used in tests.
+// If you move this function you'll have to change tests
+func crashyHandler(w http.ResponseWriter, r *http.Request) {
+	c := make(chan int)
+	close(c)
+	c <- 1
+}
+
+type _recurse struct {
+	Recurse *_recurse
+}
+
+const (
+	unhandled = true
+	handled   = false
+)
+
+var testAPIKey = "166f5ad3590596f9aa8d601ea89af845"
+
+type logger struct{ msg string }
+
+func (l *logger) Printf(format string, v ...interface{}) { l.msg = format }
+
+// setup sets up a simple sessionTracker and returns a test event server for receiving the event payloads.
+// report payloads published to ts.URL will be put on the returned channel
```

Comment:
Afaict `ts` doesn't have any context here. It might be worth either naming your return values (so that the server is named `ts`), or changing the comment to something like '...published to the returned server's URL'.